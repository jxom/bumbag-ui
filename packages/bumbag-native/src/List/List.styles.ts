import { VirtualizedList } from 'react-native';
import { styled } from '../styled';
import { theme } from '../utils/theme';

export const StyledListFlat = styled.FlatList`
  ${theme('native.List.Flat', 'styles.base')};
`;

export const StyledListSection = styled.SectionList`
  ${theme('native.List.Section', 'styles.base')};
`;

export const StyledListVirtualized = styled(VirtualizedList)`
  ${theme('native.List.Virtualized', 'styles.base')};
`;
