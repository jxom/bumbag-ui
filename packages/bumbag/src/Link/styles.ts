import { cssClass } from '../styled';
import { palette, theme, fontWeight } from '../utils';

export const Link = (styleProps) => cssClass`
  color: ${palette('primary', { dark: 'primary300' })(styleProps)};
  fill: ${palette('primary', { dark: 'primary300' })(styleProps)};
  cursor: pointer;
  font-weight: ${fontWeight('semibold')(styleProps)};
  text-decoration: none;

  &:hover {
    text-decoration: underline;

    & {
      ${theme(styleProps.themeKey, `styles.hover`)(styleProps)};
    }
  }

  &:focus {
    & {
      ${theme(styleProps.themeKey, `styles.focus`)(styleProps)};
    }
  }

  & .bb-Icon {
    vertical-align: -0.125em;
  }

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
