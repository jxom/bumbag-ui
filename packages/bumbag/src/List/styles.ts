import { cssClass } from '../styled';
import { space, theme } from '../utils';

export const List = (styleProps) => cssClass`
  list-style: unset;
  list-style-type: none;

  & & {
    margin-left: ${space(4)(styleProps)}rem;
  }

  & li {
    &:not(:last-child) {
      margin-bottom: ${space(1)(styleProps)}rem;
    }
  }

  & {
    ${styleProps.isOrdered && getOrderedProperties(styleProps)};
  }

  & {
    ${styleProps.orientation === 'horizontal' && getHorizontalProperties(styleProps)};
  }

  & {
    ${theme(styleProps.themeKey, 'styles.base')(styleProps)};
  }
`;

export const ListItem = (styleProps) => cssClass`
  & .fp-Icon {
    vertical-align: -0.125em;
  }

  & {
    ${theme(styleProps.themeKey, 'styles.base')(styleProps)};
  }
`;

export const getOrderedProperties = (styleProps) => cssClass`
  list-style-type: decimal;

  & & {
    list-style-type: lower-alpha;
  }

  & & & {
    list-style-type: lower-roman;
  }

  & {
    ${theme(styleProps.themeKey, 'styles.ordered')(styleProps)};
  }
`;

export const getHorizontalProperties = (styleProps) => cssClass`
  &&& li {
    display: inline-block;
    margin-bottom: unset;

    &:not(:last-child) {
      margin-right: 1rem;
    }
  }

  & {
    ${theme(styleProps.themeKey, 'styles.horizontal')(styleProps)};
  }
`;
