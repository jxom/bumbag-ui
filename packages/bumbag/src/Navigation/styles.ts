import { cssClass } from '../styled';
import { theme } from '../utils';

export const Navigation = (styleProps) => cssClass`
  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
