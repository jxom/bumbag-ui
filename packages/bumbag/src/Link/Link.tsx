import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './Link.styles';

export type LocalLinkProps = {};
export type LinkProps = BoxProps & React.AnchorHTMLAttributes<any> & LocalLinkProps;

const useProps = createHook<LinkProps>(
  (props, { themeKey }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.Link,
      styleProps: props,
      themeKey,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Link' }
);

export const Link = createComponent<LinkProps>(
  (props) => {
    const linkProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: linkProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Link',
    },
    defaultProps: {
      use: 'a',
    },
    themeKey: 'Link',
  }
);
