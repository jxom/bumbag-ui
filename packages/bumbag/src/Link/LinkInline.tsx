import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './styles';

export type LocalLinkInlineProps = {};
export type LinkInlineProps = BoxProps & React.LinkHTMLAttributes<any> & LocalLinkInlineProps;

const useProps = createHook<LinkInlineProps>(
  (props, { themeKey, themeKeyOverride }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.LinkInline,
      styleProps: props,
      themeKey,
      themeKeyOverride,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Link.Inline' }
);

export const LinkInline = createComponent<LinkInlineProps>(
  (props) => {
    const linkInlineProps = useProps(props);
    return createElement({
      children: props.children,
      component: ReakitBox,
      use: props.use,
      htmlProps: linkInlineProps,
    });
  },
  {
    attach: {
      useProps,
      displayName: 'Link.Inline',
    },
    defaultProps: {
      use: 'a',
    },
    themeKey: 'Link.Inline',
  }
);
