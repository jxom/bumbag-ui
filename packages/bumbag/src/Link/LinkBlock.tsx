import { Box as ReakitBox } from 'reakit';

import { useClassName, createComponent, createElement, createHook } from '../utils';
import { Box, BoxProps } from '../Box';

import * as styles from './Link.styles';

export type LocalLinkBlockProps = {};
export type LinkBlockProps = BoxProps & React.AnchorHTMLAttributes<any> & LocalLinkBlockProps;

const useProps = createHook<LinkBlockProps>(
  (props, { themeKey }) => {
    const boxProps = Box.useProps(props);

    const className = useClassName({
      style: styles.LinkBlock,
      styleProps: props,
      themeKey,
      prevClassName: boxProps.className,
    });

    return { ...boxProps, className };
  },
  { themeKey: 'Link.Block' }
);

export const LinkBlock = createComponent<LinkBlockProps>(
  (props) => {
    const linkBlockProps = useProps(props);
    return createElement({ children: props.children, component: ReakitBox, use: props.use, htmlProps: linkBlockProps });
  },
  {
    attach: {
      useProps,
      displayName: 'Link.Block',
    },
    defaultProps: {
      use: 'a',
    },
    themeKey: 'Link.Block',
  }
);
