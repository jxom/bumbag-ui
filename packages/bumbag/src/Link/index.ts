import * as styles from './styles';
import { Link as _Link } from './Link';
import { LinkBlock } from './LinkBlock';
import { LinkInline } from './LinkInline';

export * from './Link';
export const Link = Object.assign(_Link, {
  Block: LinkBlock,
  Inline: LinkInline,
});
export { styles as linkStyles };
