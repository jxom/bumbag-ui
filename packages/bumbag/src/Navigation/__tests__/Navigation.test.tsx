import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Navigation } from '../Navigation';
import { List } from '../../List';
import { Link } from '../../Link';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Navigation>
        <List>
          <List.Item>
            <Link href="https://google.com">Google</Link>
          </List.Item>
          <List.Item>
            <Link href="https://github.com">GitHub</Link>
          </List.Item>
        </List>
      </Navigation>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(
      <Navigation backgroundColor="primary">
        <List>
          <List.Item>
            <Link href="https://google.com">Google</Link>
          </List.Item>
          <List.Item>
            <Link href="https://github.com">GitHub</Link>
          </List.Item>
        </List>
      </Navigation>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('use', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Navigation use="div">
          <List>
            <List.Item>
              <Link href="https://google.com">Google</Link>
            </List.Item>
            <List.Item>
              <Link href="https://github.com">GitHub</Link>
            </List.Item>
          </List>
        </Navigation>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Navigation props', () => {
      const { result } = renderHook(() => Navigation.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Navigation>
          {(NavigationProps) => (
            <div {...NavigationProps}>
              <List>
                <List.Item>
                  <Link href="https://google.com">Google</Link>
                </List.Item>
                <List.Item>
                  <Link href="https://github.com">GitHub</Link>
                </List.Item>
              </List>
            </div>
          )}
        </Navigation>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('overrides', () => {
  it('Navigation.base should render correctly', () => {
    const { container } = render(
      <Navigation overrides={{ Navigation: { styles: { base: { backgroundColor: 'red' } } } }}>
        <List>
          <List.Item>
            <Link href="https://google.com">Google</Link>
          </List.Item>
          <List.Item>
            <Link href="https://github.com">GitHub</Link>
          </List.Item>
        </List>
      </Navigation>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('Navigation.base should render correctly', () => {
    const { container } = render(
      <Navigation>
        <List>
          <List.Item>
            <Link href="https://google.com">Google</Link>
          </List.Item>
          <List.Item>
            <Link href="https://github.com">GitHub</Link>
          </List.Item>
        </List>
      </Navigation>,
      {
        theme: { Navigation: { styles: { base: { backgroundColor: 'red' } } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(
      <Navigation>
        <List>
          <List.Item>
            <Link href="https://google.com">Google</Link>
          </List.Item>
          <List.Item>
            <Link href="https://github.com">GitHub</Link>
          </List.Item>
        </List>
      </Navigation>,
      {
        theme: { Navigation: { defaultProps: { className: 'test' } } },
      }
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
