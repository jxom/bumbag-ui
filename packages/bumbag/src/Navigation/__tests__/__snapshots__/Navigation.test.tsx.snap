// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition hook should return with Navigation props 1`] = `
Object {
  "className": "css-1fie8gx css-63it0p css-jwhr8m bb-Box bb-Navigation",
  "role": "navigation",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</div>
`;

exports[`composition use should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<div
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
  role="navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</div>
`;

exports[`defaultProps should render correctly for className 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<nav
  class="emotion-0 emotion-1 emotion-2 test bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;

exports[`overrides Navigation.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<nav
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<nav
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-2 {
  background-color: var(--bb-palette-primary, #000000)!important;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<nav
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-5 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-5 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-5 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-5 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-5 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;

exports[`theming Navigation.base should render correctly 1`] = `
.emotion-0 {
  background-color: var(--bb-palette-red, #ff0000);
}

.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-3 {
  list-style: unset;
  list-style-type: none;
}

.emotion-3 .emotion-3 {
  margin-left: 1rem;
}

.emotion-3 li:not(:last-child) {
  margin-bottom: 0.25rem;
}

.emotion-6 .bb-Icon {
  vertical-align: -0.125em;
}

.emotion-9 {
  color: var(--bb-palette-primary, #000000);
  fill: var(--bb-palette-primary, #000000);
  cursor: pointer;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 .bb-Icon {
  vertical-align: -0.125em;
}

<nav
  class="emotion-0 emotion-1 emotion-2 bb-Box bb-Navigation"
>
  <ul
    class="emotion-3 emotion-1 emotion-2 bb-Box bb-List"
  >
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://google.com"
      >
        Google
      </a>
    </li>
    <li
      class="emotion-6 emotion-1 emotion-2 bb-Box bb-ListItem"
    >
      <a
        class="emotion-9 emotion-1 emotion-2 bb-Box bb-Link"
        href="https://github.com"
      >
        GitHub
      </a>
    </li>
  </ul>
</nav>
`;
